File: 3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.py
class Solution(object):
    def maxDifference(self, s):
        """
        :type s: str
        :rtype: int
        """
        map=Counter(s);
        min_even=float('inf')
        max_odd=float('-inf')
        for i in map.values():
            if(i%2==0):
                min_even=min(min_even,i)
            else:
                max_odd=max(max_odd,i)
        return max_odd-min_even

File: 3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.java
class Solution {
    public int maxDifference(String s) {
        HashMap<Character,Integer>map=new HashMap<>();
        int n=s.length();
        for(int i=0;i<n;i++){
            char ch=s.charAt(i);
            map.put(ch,map.getOrDefault(ch,0)+1);
        }
        int min_even=Integer.MAX_VALUE;
        int max_odd=Integer.MIN_VALUE;
        for(int i:map.values()){
            if(i%2==0){
                min_even=Math.min(min_even,i);
            }
            else{
                max_odd=Math.max(max_odd,i);
            }
        }
        return max_odd-min_even;
        
    }
}